src/include/libpbs.h

```diff
@@ -213,6 +222,8 @@ struct batch_reply
 		struct brp_select *brp_select; /* select replies */
 		pbs_list_head brp_status; /* status (svr) replies */
 		struct batch_status *brp_statc; /* status (cmd) replies) */
+		pbs_list_head brp_delstat; /* status (deleted jobs) replies) */
```

Comment:
We don't need this and `brp_deletejobstat` struct, use `batch_deljob_status` on server side also while creating replies.

Reply:
Please explain, how can we use the same struct. It's not feasible in req_delete() function. Its possible in req_stat() & req_select()

There is one difference between req_stat and req_delete. In req_stat(), all the select records are added within the same function, so they are maintaining the next points in pselx and passing the same to each call. 
 
But in case of req_delete, the addition of delete status records needs to be added from 3 different functions, like req_delete(), req_delete2() and post_delete_mom1().

Reply:
I doesn't matter how many func its adding record as there is only one func `update_deletejob_stat()` - who is adding record in this list. Anyways its fairly simple to use batch_deljob_status and blp_delstatc instead this, u can do like this:
preply = &preq->rq_reply; // preq is passed as arugment
newstat = malloc(batch_deljob_status)
fill newstat with its name and code
newstat->next = preply->brp_delstatc;
preply->brp_delstatc = newstat;

Reply:
You should also think about encoding and free this list part. If the "preply->brp_delstatc" also holds the lastly added struct, how do you traverse to firstly added struct(remember it is a single linked list)? You are having only the pointer to the last struct. We need head pointer at two places, One at encoding operating where we are actually sending the records and another place at freeing the records after seeing(or at failure).

Reply:
@bremanandjk You already have func called `pbs_delstatfree()` to free batch_deljob_status structure, and traversing will be same as how you are traversing in `pbs_delstatfree()`

> What is the benefit we are achieving by we get by doing all these code changes, do you think, this much refactoring is worth to get the final outcome. I think this will simply delay this qdel optimization work without much benefit out of it.

Yes, I strongly believe that doing this change will be beneficial in future and now also (as you will have less new structures and pointers in code to maintain + code will be much more easily understandable)

Reply:
I have pushed the changes as per the suggestion.